### PR TITLE
GlslSandbox class syntax

### PR DESCRIPTION
**Pull request by marioecg**
I'm making a PR to port `GlslSandbox` function to an ES6 class.

In `example.js` enable importing the `GlslSandbox` class and you'll see the canvas completely in black 😅. I'd love help to know where to start debugging this.

### CODE REVIEW

File: class.js [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+
+        this.lastTime = this.time;
+        this.frame++;
+    }
+
+    updateBuffers() {
+        // Buffers
+        for (let i = 0, il = this.buffers.length; i < il; i++) {
+            let b = this.buffers[i];
+            if (db.width <= 1.0 && db.height <= 1.0)
```
Comment (patriciogonzalezvivo) [inline class.js]: There is a fix that needs to be update here https://github.com/patriciogonzalezvivo/glsl-sandbox/commit/f3bfc7c2cd9a30df7da9672418847a4a02bc232d

**Comment by patriciogonzalezvivo** [comment]
if works as expected, let's replace index for it. Also please make sure that changes made after you branch are considered.
Gracias Mario!

**Comment by marioecg** [comment]
I replaced `index.js` with the class structure and it works like a charm. The issue we were having before was that the main uniforms object wasn't being added when creating new shader materials, so now they're passed through the `createShaderMaterial` function like so:

``let material = createShaderMaterial(`#define BUFFER_${index}\n${frag_src}`, this.uniforms);``

I also updated with the fix from above. Let me know what you think!

**Comment by patriciogonzalezvivo** [comment]
Fantastic job! Thank you!

**Pull request merged**